integration_tests/src/lib.rs

```diff
@@ -2570,6 +2596,370 @@ mod test {
+        volume
+            .add_subvolume_create_guest(
+                new_opts,
+                volume::RegionExtentInfo {
+                    block_size: BLOCK_SIZE as u64,
+                    blocks_per_extent: test_downstairs_set.blocks_per_extent(),
+                    extent_count: test_downstairs_set.extent_count(),
+                },
+                3,
+                None,
+            )
+            .await?;
+
+        volume.activate().await?;
+
+        // Verify our volume still has the random data we wrote.
+        let volume_size = volume.total_size().await? as usize;
+        let mut buffer = Buffer::new(volume_size / BLOCK_SIZE, BLOCK_SIZE);
+        volume
+            .read(Block::new(0, BLOCK_SIZE.trailing_zeros()), &mut buffer)
+            .await?;
+
+        assert_eq!(&buffer[BLOCK_SIZE..], &random_buffer[BLOCK_SIZE..]);
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn integration_test_clone_diff_ec() -> Result<()> {
+        // Test downstairs replacement.
+        // Verify different extent count will fail.
+
+        let mut ds_one = TestDownstairs::new(
+            "127.0.0.1".parse().unwrap(),
+            true,
+            true,
+            5,
+            3,
+            false,
+            Backend::RawFile,
+        )
+        .await
+        .unwrap();
+
+        ds_one.reboot_read_only().await.unwrap();
+        let clone_source = ds_one.repair_address().await;
+
+        let mut ds_two = TestDownstairs::new(
+            "127.0.0.1".parse().unwrap(),
+            true,
+            true,
+            5,
+            2, // <- Different than above
+            false,
+            Backend::RawFile,
+        )
+        .await
+        .unwrap();
+
+        assert!(ds_two.reboot_clone(clone_source).await.is_err());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn integration_test_clone_diff_es() -> Result<()> {
+        // Test downstairs replacement.
+        // Verify different extent size will fail.
+
+        let mut ds_one = TestDownstairs::new(
+            "127.0.0.1".parse().unwrap(),
+            true,
+            true,
+            9,
+            2,
+            false,
+            Backend::RawFile,
+        )
+        .await
+        .unwrap();
+
+        ds_one.reboot_read_only().await.unwrap();
+        let clone_source = ds_one.repair_address().await;
+
+        let mut ds_two = TestDownstairs::new(
+            "127.0.0.1".parse().unwrap(),
+            true,
+            true,
+            5, // <- Different than above
+            2,
+            false,
+            Backend::RawFile,
+        )
+        .await
+        .unwrap();
+
+        assert!(ds_two.reboot_clone(clone_source).await.is_err());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn integration_test_clone_not_ro() -> Result<()> {
+        // Test downstairs replacement.
+        // Verify you can't replace from a RW downstairs
```

Comment:
s/replace/clone/

Reply:
Fixed in 198b0ca
Also, cleaned up a few more comments for a similar violation.